File: vdp-ruby/test/vctc/consumer_rules_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class ConsumerRulesTest < Test::Unit::TestCase
  def setup
    @visa_api_client = VisaAPIClient.new
    @config = YAML.load_file('configuration.yml')
    @cardRegisterData ='''{
      "primaryAccountNumber": '''+ @config['vctcTestPan'] + '''
    }'''
  end
  
  def test_registerACard
    base_uri = 'vctc/'
    resource_path = 'customerrules/v1/consumertransactioncontrols'
    response_code = @visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "Register a Card test", "post", @cardRegisterData)
    assert_equal(true, (response_code == "200" || response_code == "201"), "Register a Card test failed")
  end
end

File: vdp-node/test/vctc/consumerrulestest.js
var request = require('request');
var fs = require('fs');
var config = require('../../config/configuration.json');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var assert = require('chai').assert;
var randomstring = require('randomstring');

var req = request.defaults();
var userId = config.userId ;
var password = config.password;
var keyFile = config.key;
var certificateFile = config.cert;

describe('Visa Consumer Transaction Controls Consumer Rules Test', function() {
	var visaAPIClient = new VisaAPIClient();
	var cardRegisterData = JSON.stringify({
		  "primaryAccountNumber": config.vctcTestPan
	});
	
	it('Register A Card Test',function(done) {
		this.timeout(10000);
		var baseUri = 'vctc/';
		var resourcePath = 'customerrules/v1/consumertransactioncontrols';
		visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, cardRegisterData, 'POST', {}, 
		function(err, responseCode) {
		    if(!err) {
		    	assert.equal(responseCode, 200);
		    } else {
		        assert(false);
		    }
		    done();
		});
	});
});
